### PR TITLE
[t2CharStringPen] add "roundTolerance" option

### PR DESCRIPTION
**Pull request by anthrotype**
I went for the easy route, and simply added a `roundTolerance` argument to the `T2CharStringPen` as @behdad was discussing in #769.

I personally think that doing more clever things would be a bit beyond the scope of fontTools.
But I'm happy to be convinced otherwise ;)

PTAL, thanks

### CODE REVIEW

**Comment by justvanrossum** [comment]
I think the default should remain rounding behavior and that the new behavior should be explicitly turned on.

**Comment by anthrotype** [comment]
Yeah, initially I also thought it'd better to keep the current default, but then got carried away :)
I'll restore the default rounding.

Shall I assert the tolerance is comprised between 0 and 0.5? Values between 0.5 and 1.0 don't make much sense..

**Comment by behdad** [comment]
No, please don't assert tolerance.  I like to be able to say "UPEM * .005" for example and that should work.  Assertions are for other things: for when the input is wrong.  Which is not the case here.

**Comment by anthrotype** [comment]
I'm confused. I'm using the built in round() to convert float to int. The maximum tolerance (which Indefine as the max absolute difference between the rounded and unrounded values) is 0.5, and couldn't be greater.. Am I'm missing something?

File: Lib/fontTools/pens/t2CharStringPen.py [hunk 1]
```diff
@@ -77,7 +76,7 @@ def _round(number):
         # return rounded integer if the tolerance is 0.5, or if the absolute
         # difference between the original float and the rounded integer is
         # within the tolerance
-        if tolerance == .5 or fabs(rounded - number) <= tolerance:
+        if tolerance == .5 or abs(rounded - number) <= tolerance:
```
Comment (behdad) [inline Lib/fontTools/pens/t2CharStringPen.py]: Just make this "tolerance >= .5"

**Comment by behdad** [comment]
> the max absolute difference between the rounded and unrounded values) is 0.5

Correct.  Which means rounding is an acceptable solution for any user-specified tolerance >= .5.  A tolerance of 1.5 is NOT by any means an error.

**Comment by anthrotype** [comment]
Ok, I got it. Thanks

**Pull request merged**